### ISSUE TITLE
Add GPS co-ordinates into data export

### ISSUE THREAD

**Issue by dave-mills** [stats4sd/tape-data-system#76]
The TAPE team need access to the GPS data collected during the survey.

**Comment by dan-tang-ssd** [comment]
Please find my assessment and proposed solution as below for comments, Thanks.

---

**Assessment**

- check data extraction file
  - Data Dictionary excel sheet, in Main Survey > farm_info, there are 6 ODK variables
    - respondent_name,respondent_phone,latitude,longitude,altitude,accuracy
  - Main_Survey excel sheet, found 3 columns
    - respondent_name, respondent_phone, gps_loc
  
 - GPS co-ordinates are not stored in main_surveys table
  
 - GPS co-ordinates are not stored in entity_values table
  
- GPS co-ordinates are only stored in submissions.content now
  - farm_info\gps_loc\coordinates
  - farm_info\gps_loc\properties\accuracy

---

**Proposed solution**

 - main_surveys table, add new columns latitude,longitude,altitude,accuracy after column gps_loc
 - develop one-time program, find GPS details from corresponding submissions.content for all main_survey records, update to database
 - perform data extraction for latest excel file with GPS coordinates
 - add the splitted columns for select_multiple columns to 3 excel sheets
 - send the finalised excel file to user

**Comment by dave-mills** [comment]
Sounds good - thanks. 2 minor things:

- No need to keep the (currently empty) gps_loc column if we're adding in the split gps columns. 
- While you're there, can you also populate the gps columns in the farms table? It may be of value to the teams to have that list of farms with updated info in the future.

**Comment by dan-tang-ssd** [comment]
>     * No need to keep the (currently empty) gps_loc column if we're adding in the split gps columns.
Sure, no problem.


>     * While you're there, can you also populate the gps columns in the farms table? It may be of value to the teams to have that list of farms with updated info in the future.

I did a checking with below SQL:
`select * from main_surveys where respondent_name like 'Tadase Dame Waktola';`

Just to double confirm... 
Is main_surveys.respondent_name = farms.identifiers\name?

I noticed there are some farms records with GPS details already (maybe we did some deployment after launch...?). Should I only populate to those farms without GPS details?

---

Screen shots:

![image](https://github.com/user-attachments/assets/7a65cf74-1315-4717-b053-dcb13badd190)

![image](https://github.com/user-attachments/assets/dff1d5ef-bcc1-4734-a578-a3f24b162fcf)
